checks if required user input files exist and if variants match

naqvia opened the pull request:
This script  performs multiple tests. It first checks to see if required vcf, multianno, intervar, and autopvs1, clinvar, submission summary, and variant_summary files exists and then also ensures the variants in the vcf file match against multianno and  autopvs1. If any of these tests fail it will report the specific issue. For example if there is a variant mismatch, it will print vcf variant that does not have an entry in the other files. This should/can be ran before running the downstream tool. 

Here is a sample run where it by default checks the `input` dir:
`perl 00-check_files.pl`

Currently it can only handle one sample at a time. Will eventually out this in a bash script to run. We don't need print statements for matches but have them for understanding and for others to follow since this is coded in perl.

Related to https://github.com/diskin-lab-chop/AutoGVP/issues/45

rjcorb commented on AutoGVP/00-file_and_variant_check.pl:
```diff
@@ -0,0 +1,150 @@
+}
+
+## check to see if input files all exist
+if (glob($dir_to_check."/*.vcf")) {
+  print "vcf file exists...\n";
+  $vcf_file_check = $dir_to_check."/*.vcf";
```
```suggestion
if (glob($dir_to_check."/*.vcf*")) {
  print "vcf file exists...\n";
  $vcf_file_check = $dir_to_check."/*.vcf*";
```

modifying in the event vcf file is compressed

rjcorb commented on AutoGVP/00-file_and_variant_check.pl:
```diff
@@ -0,0 +1,150 @@
+  die("FAIL: multianno file does not exist\n");
+}
+
+if (glob($dir_to_check."/*autopvs1.txt")) {
+  print "autopvs1 file exists...\n";
+  $autopvs1_file_check = $dir_to_check."/*autopvs1.txt";
+
+}
```
```suggestion
if (glob($dir_to_check."/*autopvs1*")) {
  print "autopvs1 file exists...\n";
  $autopvs1_file_check = $dir_to_check."/*autopvs1*";

}
```

reducing search to just "autopvs1", in case files are tsv or some other format

rjcorb commented on AutoGVP/00-file_and_variant_check.pl:
```diff
@@ -0,0 +1,150 @@
+}
+close(FIL);
+
+open(FIL,$multianno_file) || die("Cannot Open File $vcf_file");
```
```suggestion
open(FIL,$multianno_file) || die("Cannot Open File $multianno_file");
```

rjcorb commented on AutoGVP/00-file_and_variant_check.pl:
```diff
@@ -0,0 +1,150 @@
+}
+close(FIL);
+
+open(FIL,$autopvs1_file) || die("Cannot Open File $vcf_file");
```
```suggestion
open(FIL,$autopvs1_file) || die("Cannot Open File $autopvs1_file");
```

rjcorb reviewed (changes requested):
This looks good so far--I've made some code suggestions, and wonder if this can be modified to read in gzipped files as well. Also, what is the utility of this when you have input files from many different samples in the same directory? I'm thinking an argument for sample ID/file prefix will be needed to search for the right files to compare.

rjcorb commented on AutoGVP/00-file_and_variant_check.pl:
```diff
@@ -0,0 +1,157 @@
+  next if $_=~/Start/;
+
+  my @cols = split;
+  my $chr = $cols[25];
+  my $start = $cols[26];
```
The column #s will change here depending on what information is included in the multianno file. For example, in the PBTA output, the desired `position` value is actually column 151. I think this will make it difficult to match coordinates using this method, unless there is some way in perl to pull columns based on header information? I'm pretty unfamiliar with the language...

naqvia replied:
can you paste a few lines of your test file? Will try to modify code based on that...

rjcorb reviewed (changes requested):
I tested this on some PBTA samples and it failed because the column # for position in the multianno file is different from the one used in this perl script (which I assume corresponds to the correct column # in the test files). See comment below. This will have to be modified to be robust against different multianno files that will include a variable number of columns.

naqvia commented:
I modified the code for multianno parsing to instead parse via regex, so we don't parse via column #. The following if statement is key and what I added: 

`if($_=~/(chr\d+)\t(\d+)\t(rs\d+|\.)\t/)`

the format should be `chr`, `start`, and then either `rs_id` or a `.`. Delimited by a tab. If the format differs from that, this regex will not work properly. I tested it on test files and an example file from Ryan and it works as intended and I have not seen output files differ from this format.

naqvia commented:
this should be ready to be reviewed. Had to go through R scripts and make sure we resolved all conflicts. cc @rjcorb
